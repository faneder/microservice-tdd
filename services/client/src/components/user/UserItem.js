import React from "react";

import {
  TableCell,
  TableRow
} from '@material-ui/core';

const UserItem = (user) => {
  return (
    <TableRow key={user.id}>
      <TableCell align="right">{user.id}</TableCell>
      <TableCell align="right">{user.email}</TableCell>
      <TableCell align="right">{user.username}</TableCell>
      <TableCell align="right">{user.active ? 'o' : 'x'}</TableCell>
    </TableRow>
  )
};

export default UserItem;