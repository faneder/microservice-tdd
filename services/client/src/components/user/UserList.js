import React from 'react';
import PropTypes from 'prop-types';
import UserItem from './UserItem';
// @material-ui/core components
import {
  withStyles,
  Table,
  TableRow,
  TableCell,
  TableHead,
  TableBody,
  Paper
} from '@material-ui/core';

const styles = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
  },
  table: {
    minWidth: 700,
  },
});

function UserList(props) {
  const { classes } = props;

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell align="right">Name</TableCell>
            <TableCell align="right">email</TableCell>
            <TableCell align="right">username</TableCell>
            <TableCell align="right">active</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {
            props.users.map((user) => {
              return (
                <UserItem key={user.id} {...user} />
              )
            })
          }
        </TableBody>
      </Table>
    </Paper>
  );
}

UserList.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(UserList);
