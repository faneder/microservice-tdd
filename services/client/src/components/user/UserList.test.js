import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import UserList from "./UserList";
import UserItem from "./UserItem";

import {
  Table,
  TableHead,
  TableBody,
} from '@material-ui/core';

const users = [
  {
    'id': 1,
    'active': true,
    'email': 'eder@gmail.com',
    'username': 'eder'
  },
  {
    'id': 2,
    'active': true,
    'email': 'ping@ping.org',
    'username': 'ping'
  }
];

test('UserList renders properly', () => {
  const wrapper = mount(<UserList users={users} />);

  expect(wrapper.find(Table)).toHaveLength(1);
  expect(wrapper.find(TableHead)).toHaveLength(1);
  expect(wrapper.find(TableBody)).toHaveLength(1);
});

test('It renders 2 UserItems when 2 users ', () => {
  const wrapper = mount(<UserList users={users} />);

  expect(wrapper.find(UserItem)).toHaveLength(2);
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UserList users={users}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
