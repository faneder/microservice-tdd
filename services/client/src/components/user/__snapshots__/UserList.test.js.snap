// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 UserList-root-1"
>
  <table
    className="MuiTable-root-30 UserList-table-2"
  >
    <thead
      className="MuiTableHead-root-31"
    >
      <tr
        className="MuiTableRow-root-32 MuiTableRow-head-35"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
          scope="col"
        >
          email
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
          scope="col"
        >
          username
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
          scope="col"
        >
          active
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-49"
    >
      <tr
        className="MuiTableRow-root-32"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          eder@gmail.com
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          eder
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          o
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-32"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          ping@ping.org
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          ping
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
        >
          o
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
